src/Microsoft.ML.ImageAnalytics/ImageGrayscaleTransform.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.ImageAnalytics;
+
+[assembly: LoadableClass(ImageGrayscaleTransform.Summary, typeof(ImageGrayscaleTransform), typeof(ImageGrayscaleTransform.Arguments), typeof(SignatureDataTransform),
+    ImageGrayscaleTransform.UserName, "ImageGrayscaleTransform", "ImageGrayscale")]
+
+[assembly: LoadableClass(ImageGrayscaleTransform.Summary, typeof(ImageGrayscaleTransform), null, typeof(SignatureLoadDataTransform),
+    ImageGrayscaleTransform.UserName, ImageGrayscaleTransform.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.ImageAnalytics
+{
+    // REVIEW: Rewrite as LambdaTransform to simplify.
+    // REVIEW: Should it be separate transform or part of ImageResizerTransform?
+    /// <summary>
+    /// Transform which takes one or many columns of <see cref="ImageType"/> type in IDataView and
+    /// convert them to greyscale representation of the same image.
+    /// </summary>
+    public sealed class ImageGrayscaleTransform : OneToOneTransformBase
+    {
+        public sealed class Column : OneToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public class Arguments : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+        }
+
+        internal const string Summary = "Convert image into grayscale.";
+
+        internal const string UserName = "Image Greyscale Transform";
+        public const string LoaderSignature = "ImageGrayscaleTransform";
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "IMGGREY ",
```

Comment:
>IMGGREY  [](start = 33, length = 8)

You had a signature of `IMGLOADT` for that. Why is this not `IMGGREYT`?

Also: is it intentional that this be `GREY`?

If models already exist with this signature I guess we have to keep it, not sure if that's the case though since you didn't bump the version. :) (I'm not suggesting you bump the version to be clear.)

Reply:
English are so confusing, grey and gray is apparent to have same meaning, but my brain apparently throw coin during typing and at some point I had code with both spelling all across code. Which I tried to unified, but this one slip from my attention.
This is a new transform which we never had before, and I add it "just in case", and "maybe it will prove useful".  So i think it make sense to start from version 1.

---
In reply to: [205265843](https://github.com/dotnet/machinelearning/pull/528#discussion_r205265843) [](ancestors = 205265843)

Reply:
Oh sure @Ivanidzo4ka grey and gray are synonymous, though elsewhere you use gray. Also my impression is that we generally favor American spellings (gray) of words vs. British ones (grey).

...Even though I was brought up to use "grey," and had to correct myself since later in my life people thought I was weird. So now I hypercorrect, and you reap the benefits. :D

---
In reply to: [205597316](https://github.com/dotnet/machinelearning/pull/528#discussion_r205597316) [](ancestors = 205597316,205265843)